packages/microsoft_dnsserver/data_stream/audit/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,429 @@
+---
+description: Pipeline for processing DNS Server Audit logs
+processors:
+  - set:
+      field: ecs.version
+      value: 8.11.0
+      tag: set_ecs_version
+  - set:
+      field: event.kind
+      value: event
+      tag: set_event_kind
+  
+  - set:
+      field: event.category
+      value: [configuration]
+      if: ctx.winlog?.task != null && ctx.winlog.task == "Configuration"
```

```suggestion
      if: ctx.winlog?.task == "Configuration"
```
You can possibly combine both conditions

---

```
if: ctx.winlog?.task == "Configuration"
```
This condition covers the possibility that `ctx.winlog` exists but `ctx.winlog.task` is not present? If so, I will update the whole pipeline. Just asking because I see many integrations having both checks.

---

You can combine these; `ctx.winlog?.task` resolves to `null` if `ctx.winlog` or `ctx.winlog.task` are `null`/absent, and `null` does not equal "Configuration", so the check is valid.